app/assets/src/components/views/compare/SamplesHeatmapVis.jsx

```diff
@@ -211,6 +216,24 @@ class SamplesHeatmapVis extends React.Component {
     logAnalyticsEvent("SamplesHeatmapVis_metadata-node_hovered", metadata);
   };
 
+  handleRowGroupEnter = (rowGroup, rect, minTop) => {
+    const tooltipWidth = Math.max(MAX_ROW_GROUP_TOOLTIP_WIDTH, rect.width / 2);
+    this.setState({
+      rowGroupLegend: {
+        label: `Genus: ${rowGroup.genusName || "Unknown"}`,
+        tooltipLocation: {
+          left: rect.left + rect.width / 2 - tooltipWidth / 2, // center
+          top: Math.max(minTop, rect.top),
+          width: tooltipWidth,
```

Comment:
@MarkAZhang and @tfrcarvalho: @happyimadesignr  and I would like to know whether you can think of any way to center the tooltip in the row label column while the tooltip's size is dynamic based on the text? 

All I can think of would be to render all the tooltips, keep them hidden with CSS, and show them only on hover. Thoughts?

Reply:
You can render an invisible placeholder that has the width of the label column (width of rect) and left 0 from the left of the enclosing rectangle. And then align the tooltip inside the placeholder.

Reply:
You can do, in `RowGroupLegend`
```
{{
  transform: "translate(-50%)",
  left: tooltipLocation.left + 4, // offset because of right-alignment of labels
  top: tooltipLocation.top - 44, // depends on heights defined in CSS
}}
```

And then here, just do `left: rect.left + rect.width / 2`.

`translate(-50%)` translate by 50% of the dynamic width of the tooltip, which is what you want.

Percentage when using `translate` refers to the current element's width, whereas most CSS rules (like left, top, margin) refer to the parent element's width. The difference is a little confusing to remember, but helpful for cases like this.

Reply:
Translate  / transform works perfectly! Bravo @MarkAZhang . You are making my CSS knowledge feel out of date!